### PR TITLE
Adds http_client_request_test.dart

### PR DESCRIPTION
**Pull request by zanderso**

### CODE REVIEW

**Review by bkonyi — APPROVED** [review]

File: test/http_client_request_test.dart [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      });
+    });
+    return request.done;
+  });
+}
+
+Future<Null> testBadResponseAdd() async {
+  await testClientRequest((request) {
+    request.contentLength = 0;
+    request.add([0]);
```
Comment (kevmoo) [inline test/http_client_request_test.dart:49]: FYI: these tests don't "work"

Comment out the `add` call here (which should cause the error) and the test passes.

Comment (kevmoo) [reply test/http_client_request_test.dart:49]: @zanderso – these need to be reviewed carefully –

Comment (zanderso) [reply test/http_client_request_test.dart:49]: @kevmoo Thanks for pointing this out. It's a shame these tests weren't carefully reviewed when they were first submitted.

Comment (kevmoo) [reply test/http_client_request_test.dart:49]: agreed!

**Pull request merged**